File: java-dates/src/test/java/com/baeldung/datetime/UsePeriodUnitTest.java
package com.baeldung.datetime;

import java.time.LocalDate;
import java.time.Period;

import org.junit.Assert;
import org.junit.Test;

public class UsePeriodUnitTest {
    UsePeriod usingPeriod = new UsePeriod();

    @Test
    public void givenPeriodAndLocalDate_thenCalculateModifiedDate() {
        Period period = Period.ofDays(1);
        LocalDate localDate = LocalDate.parse("2007-05-10");
        Assert.assertEquals(localDate.plusDays(1), usingPeriod.modifyDates(localDate, period));
    }

    @Test
    public void givenDates_thenGetPeriod() {
        LocalDate localDate1 = LocalDate.parse("2007-05-10");
        LocalDate localDate2 = LocalDate.parse("2007-05-15");

        Assert.assertEquals(Period.ofDays(5), usingPeriod.getDifferenceBetweenDates(localDate1, localDate2));
    }
}

File: core-kotlin-modules/core-kotlin-2/src/test/kotlin/com/baeldung/dates/datetime/UsePeriodUnitTest.kt
package com.baeldung.kotlin.datetime

import com.baeldung.dates.datetime.UsePeriod
import org.junit.Assert
import org.junit.Test
import java.time.LocalDate
import java.time.Period

class UsePeriodUnitTest {

    var usingPeriod = UsePeriod()

    @Test
    fun givenPeriodAndLocalDate_thenCalculateModifiedDate() {
        val period = Period.ofDays(1)
        val localDate = LocalDate.parse("2007-05-10")
        Assert.assertEquals(localDate.plusDays(1), usingPeriod.modifyDates(localDate, period))
    }

    @Test
    fun givenDates_thenGetPeriod() {
        val localDate1 = LocalDate.parse("2007-05-10")
        val localDate2 = LocalDate.parse("2007-05-15")

        Assert.assertEquals(Period.ofDays(5), usingPeriod.getDifferenceBetweenDates(localDate1, localDate2))
    }
}